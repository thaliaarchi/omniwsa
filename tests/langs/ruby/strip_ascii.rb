puts "Ruby #{RUBY_VERSION}"
puts "ASCII whitespace characters for String#strip:"
(0..255).each do |c|
  s = c.chr(Encoding::ASCII_8BIT)
  both = "#{s}x#{s}"
  stripped = both.strip
  if stripped != both
    direction = if stripped == "x"
      "on both sides"
    elsif stripped == "x#{s}"
      "only on the left"
    elsif stripped == "#{s}x"
      "only on the right"
    else
      "BUG"
    end
    puts "0x#{c.to_s(16).rjust(2, '0')} (#{s.inspect}) is stripped #{direction}"
  end
end
