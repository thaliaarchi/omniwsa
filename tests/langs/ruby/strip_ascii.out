Ruby 3.3.6
ASCII whitespace characters for String#strip:
0x00 ("\x00") is stripped on both sides
0x09 ("\t") is stripped on both sides
0x0a ("\n") is stripped on both sides
0x0b ("\v") is stripped on both sides
0x0c ("\f") is stripped on both sides
0x0d ("\r") is stripped on both sides
0x20 (" ") is stripped on both sides
